### ISSUE TITLE
Feature: expose the ngDefaultControl

### ISSUE THREAD

**Issue by gogoout** [NativeScript/nativescript-angular#1372]
Currently all the form controls have implemented the `ControlValueAccessor`. While if the users wants to wrap the form controls themselves, it will be quite redundant. They have to implement the ControlValueAccessor all over again.

In [here](https://stackoverflow.com/questions/47377411/wrapping-a-formcontrol-in-angular-2), the way is much more easier thanks to the `ngDefaultControl` which Angular exposed.
```
@Component({
  selector: 'my-wrapper',
  template: '<input ngDefaultControl>',
  providers: [
    {
      provide: NG_VALUE_ACCESSOR,
      useExisting: forwardRef(() => NumberInputComponent),
      multi: true,
    },
  ],
})
export class MyWrapper implements ControlValueAccessor {
  @ViewChild(DefaultValueAccessor) private valueAccessor: DefaultValueAccessor;

  writeValue(obj: any) {
    this.valueAccessor.writeValue(obj);
  }

  registerOnChange(fn: any) {
    this.valueAccessor.registerOnChange(fn);
  }

  registerOnTouched(fn: any) {
    this.valueAccessor.registerOnTouched(fn);
  }

  setDisabledState(isDisabled: boolean) {
    this.valueAccessor.setDisabledState(isDisabled);
  }
}
```
The way doesn't work in the nativescript-angular because the `ngDefaultControl` is not exposed in [here](nativescript-angular/nativescript-angular/forms/value-accessors/text-value-accessor.ts).
By simply put the `ngDefaultControl` selector in all the value-accessor. We can implement the same feature.

**Comment by gms1** [comment]
If I understand that correctly, you want to use the 'ngDefaultControl' directive for NOT binding an element to this 'DefaultValueAccessor':
https://github.com/angular/angular/blob/master/packages/forms/src/directives/default_value_accessor.ts
but you still like to get a reference to this 'DefaultValueAccessor' using "@ViewChild (DefaultValueAccessor)"?

**Comment by gogoout** [comment]
No, I believe that's not what I want. I'll try to explain in detail.
The thing is I want to wrap a TextView(or Switch/ListPicker/... etc) into a custom component kind like this:

```js
@Component({
    selector: 'ag-float-input',
    moduleId: module.id,
    template: '<TextField  ngDefaultControl></TextField>'
    providers: [
        {
            provide: NG_VALUE_ACCESSOR,
            useExisting: forwardRef(() => FloatInputComponent),
            multi: true,
        },
    ],
})
export class FloatInputComponent implements ControlValueAccessor, AfterViewInit {
   @ViewChild(TextValueAccessor) private valueAccessor: TextValueAccessor;

   public writeValue(value: string): void {
        this.valueAccessor.writeValue(value);
    }

  ...
}
```

The problem here is that I can't get the `TextValueAccessor` directive here  because the selector of the direcitve is 

from https://github.com/NativeScript/nativescript-angular/blob/master/nativescript-angular/forms/value-accessors/text-value-accessor.ts
```js
"TextField[ngModel],TextField[formControlName],TextField[formControl]," +
        "textField[ngModel],textField[formControlName],textField[formControl]," +
        "textfield[ngModel],textfield[formControlName],textfield[formControl]," +
        "text-field[ngModel],text-field[formControlName],text-field[formControl]," +
         ....
```

What I want to do is adding a selector to the Nativescript selector so that I can get a reference to the `TextValueAccessor `

So the selector will turn into something like this:
```js
"TextField[ngModel],TextField[formControlName],TextField[formControl], TextField[ngDefaultControl]," +
        "textField[ngModel],textField[formControlName],textField[formControl],textField[ngDefaultControl]," +
        "textfield[ngModel],textfield[formControlName],textfield[formControl],textfield[ngDefaultControl]," +
        "text-field[ngModel],text-field[formControlName],text-field[formControl],text-field[ngDefaultControl]," +
```

Well, to be honest, I'm still new to the Nativescript and Angular so I'm not sure if this is going to work. But the idea is to expose a way to get the Nativescripts' `TextValueAccessor`(and also other value accessors) so that we can easily wrap the native form component into a custom one.

**Comment by gms1** [comment]
ok, but the actual name of the attribute selector is almost completely irrelevant for this use case.

For example, if the TextValueAccessor would have the '[nsDefaultTextControl]' attribute declared and you would use that attribute for a TextField element, you should be able to resolve the TextValueAccessor instance by using @ViewChild(TextValueAccessor).

On the other hand, if the TextValueAccessor AND the DefaultValueAccessor would have the same '[ngDefaultControl]' attribute declared, angular should find two different value accessors for the same css selector and should throw an error

see https://github.com/angular/angular/issues/3011

**Comment by gogoout** [comment]
Thanks, that make sense.
